### DISCUSSION TITLE
audio file uploads

### DISCUSSION THREAD

**Discussion by joskezelensky** [open-webui/open-webui#3194] [General]
# Bug Report

## Description

**Bug Summary:**
[Provide a brief but clear summary of the bug]
Everytime i upload an audio file in open-webui, it fails to recognize and gets uploaded as plaintext
**Steps to Reproduce:**
[Outline the steps to reproduce the bug. Be as detailed as possible.]
upload an audio file
**Expected Behavior:**
[Describe what you expected to happen.]
audio file gets recognized as an audio file
**Actual Behavior:**
[Describe what actually happened.]
audio file gets uploaded as plaintext 
## Environment

- **Open WebUI Version:** v0.3.2
- **Ollama (if applicable):** 0.1.44

- **Operating System:** [e.g., Windows 10, macOS Big Sur, Ubuntu 20.04] windows 11
- **Browser (if applicable):** [e.g., Chrome 100.0, Firefox 98.0] latest [thorium](https://thorium.rocks/)

## Reproduction Details
upload any audiofile
**Confirmation:**

- [X] I have read and followed all the instructions provided in the README.md.
- [X] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]
not applicable, but here is python error
ERROR:apps.rag.main:Timeout reached while detecting encoding for C:\Windows\System32\open-webui\backend\data/uploads/upload.opus
Traceback (most recent call last):
  File "C:\Program Files\Python312\Lib\site-packages\langchain_community\document_loaders\text.py", line 43, in lazy_load
    text = f.read()
           ^^^^^^^^
  File "C:\Program Files\Python312\Lib\encodings\cp1252.py", line 23, in decode
    return codecs.charmap_decode(input,self.errors,decoding_table)[0]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
UnicodeDecodeError: 'charmap' codec can't decode byte 0x8d in position 418: character maps to <undefined>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Program Files\Python312\Lib\site-packages\langchain_community\document_loaders\helpers.py", line 43, in detect_file_encodings
    encodings = future.result(timeout=timeout)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\Python312\Lib\concurrent\futures\_base.py", line 458, in result
    raise TimeoutError()
TimeoutError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Windows\System32\open-webui\backend\apps\rag\main.py", line 1068, in store_doc
    data = loader.load()
           ^^^^^^^^^^^^^
  File "C:\Program Files\Python312\Lib\site-packages\langchain_core\document_loaders\base.py", line 29, in load
    return list(self.lazy_load())
           ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\Python312\Lib\site-packages\langchain_community\document_loaders\text.py", line 46, in lazy_load
    detected_encodings = detect_file_encodings(self.file_path)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\Python312\Lib\site-packages\langchain_community\document_loaders\helpers.py", line 45, in detect_file_encodings
    raise TimeoutError(
TimeoutError: Timeout reached while detecting encoding for C:\Windows\System32\open-webui\backend\data/uploads/upload.opus
**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Installation Method

[Describe the method you used to install the project, e.g., manual installation, Docker, package manager, etc.]
installed on windows from git src
## Additional Information

[Include any additional details that may help in understanding and reproducing the issue. This could include specific configurations, error messages, or anything else relevant to the bug.]
{
    "detail": "Something went wrong :/\nTimeout reached while detecting encoding for C:\\Windows\\System32\\open-webui\\backend\\data/uploads/upload.opus"
}
## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by tjbck** [comment]
Unable to reproduce the issue here, if you could share your audio file for debugging purposes that would be great!

**Reply by joskezelensky** [reply]
> Unable to reproduce the issue here, if you could share your audio file for debugging purposes that would be great!

https://files.catbox.moe/ljl6e3.aac
this is the audio (podcast) file
